features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -0,0 +1,412 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.domain.scope.RumViewManagerScope
+import com.datadog.android.rum.model.ViewEvent
+import com.datadog.android.rum.utils.forge.Configurator
+import com.datadog.tools.unit.extensions.TestConfigurationExtension
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.json.JSONObject
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+import java.util.concurrent.TimeUnit
+import kotlin.random.Random
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(TestConfigurationExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+class RumSessionEndedMetricTest {
+
+    @StringForgery
+    private lateinit var fakeSessionId: String
+
+    @Forgery
+    private lateinit var fakeStartReason: RumSessionScope.StartReason
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Test
+    fun `M create attributes with session ended metric type W toMetricAttributes`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        assertEquals(SessionEndedMetric.METRIC_TYPE_VALUE, attributes[SessionEndedMetric.METRIC_TYPE_KEY])
```

Comment:
very minor: in most cases we are using `assertThat(...).isEqualTo(...)` instead of `assertEquals`. but this is really a minor thing. Overall, `assertThat` usage can be applied to all the call sites like `assertEquals`, `assertTrue`, etc. call sites below. It gives a set of more specific assertions depending on the argument type.

```suggestion
        assertEquals(SessionEndedMetric.METRIC_TYPE_VALUE, attributes[SessionEndedMetric.METRIC_TYPE_KEY])
```

Reply:
Good point, updated